test/dialect/mysql/test_types.py

```diff
@@ -478,6 +478,18 @@ def test_float_type_compile(self, type_, sql_text):
         self.assert_compile(type_, sql_text)
 
 
+class TypeCompileUUIDMariaDBTest(fixtures.TestBase, AssertsCompiledSQL):
+    __only_on__ = "mariadb"
+
+    @testing.only_if("mariadb>=10.7")
+    def test_uuid(self):
+        self.assert_compile(sqltypes.Uuid(), "UUID")
```

Comment:
you can create a dialect here and inject a server version info into it:

```py
>>> from sqlalchemy.dialects import mysql
>>> dialect = mysql.dialect(is_mariadb=True)
>>> dialect.server_version_info = (10, 6, 5)

```

combinations to test here:

1. mariadb, version < 10.7
2. mariadb, version>= 10.7
3. mysql
4. mariadb, version >= 10.7, but native_uuid=False
5. mariadb, using sqltypes.UUID() datatype, that has to work also

see tests like https://github.com/sqlalchemy/sqlalchemy/blob/bff15276508dd47cf905cdb18ea3ae9fcae9cecb/test/dialect/mysql/test_types.py#L37 to see how to make just one test function do lots of different combinations

Reply:
Thanks, tests with combinations added for mariadb.
Not sure about mysql, because PR is only for mariadb